<a href="https://colab.research.google.com/github/mustafatmz/ayrik-sistemler-icin-ileri-olasilik/blob/master/Odev1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Soru: Uluslararası bir konferanstaki katılımcıların en az ikisinin aynı ülkeden gelme olasılığının en az %40 olması için kaç katılımcı olmalıdır?

Dünya üzerinde 206 adet ülke bulunmaktadır. İkinci kişinin ilk kişi ile aynı ülkeden gelmeme olasıklığı 205/206 olur. Aynı şekilde üçüncü kişinin diğer iki kişiden farklı bir ülkeden gelme olasılığı 204/206 olur. Bu şekilde koda dökersek;

In [8]:
from pylab import *
import numpy as np

def ayniulke(n):
  a=1
  for i in range(n):
    a=a*((206-i)/206)
  a=round(100*(1-a),4);
  return a
for n in range(50):
  print(n,ayniulke(n));

0 0
1 0.0
2 0.4854
3 1.4516
4 2.8868
5 4.7725
6 7.0838
7 9.7901
8 12.8555
9 16.2397
10 19.8992
11 23.7876
12 27.8572
13 32.0597
14 36.3472
15 40.6731
16 44.993
17 49.2654
18 53.4522
19 57.5195
20 61.4376
21 65.1815
22 68.731
23 72.0704
24 75.1887
25 78.0794
26 80.7396
27 83.1706
28 85.3764
29 87.364
30 89.1429
31 90.724
32 92.1199
33 93.344
34 94.4103
35 95.3328
36 96.1258
37 96.8028
38 97.3771
39 97.8609
40 98.2659
41 98.6026
42 98.8807
43 99.1089
44 99.2949
45 99.4455
46 99.5667
47 99.6634
48 99.7402
49 99.8007


Kod çıktısından görülebileceği üzere konferansta 15 katılımcı olması durumunda en az ikisinin aynı ülkeden gelmiş olma olasılığı %40.6731 olur.